src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +58,422 @@ class ServerFailed(Exception):
+        self.export = BasicParameter(None)          # Orterun param
+        self.processes = BasicParameter(None)       # Orterun param
+        self.env = BasicParameter(None)             # Orterun param
+
+    @property
+    def hosts(self):
+        """Hosts attribute getter."""
+        return self._hosts
+
+    @hosts.setter
+    def hosts(self, value):
+        """Hosts attribute setter
+
+        Args:
+            value (tuple): (list of hosts, workdir, slots)
+        """
+        self._hosts, workdir, slots = value
+        self.runner.processes.value = len(self._hosts)
+        self.runner.hostfile.value = write_host_file(
+            self._hosts, workdir, slots)
+        self.runner.job.server_cnt = len(self._hosts)
+
+    def get_params(self, test):
+        """Get values from the yaml file and assign them respectively
+            to the server command and the orterun command.
+
+        Args:
+            test (Test): avocado Test object
+        """
+        server_params = ["attach", "debug", "insecure", "sudo"]
+        runner_params = ["enable_recovery", "export"]
+        super(ServerManager, self).get_params(test)
+        self.runner.job.yaml_params.get_params(test)
+        self.runner.get_params(test)
+        for name in self.get_param_names():
+            if name in server_params:
+                if name == "sudo":
+                    setattr(self.runner.job, name, getattr(self, name).value)
+                else:
+                    getattr(
+                        self.runner.job, name).value = getattr(self, name).value
+            if name in runner_params:
+                getattr(self.runner, name).value = getattr(self, name).value
+
+    def run(self):
+        """Execute the runner subprocess."""
+        return self.runner.run()
+
+    def start(self, yamlfile):
+        """Start the server through the runner."""
+        self.runner.job.set_config(yamlfile)
+        self.server_clean()
+
+        # Prepare nvme storage in servers
+        if self.runner.job.yaml_params.is_nvme():
+            self.log.info("Performing nvme storage prepare in <format> mode")
+            storage_prepare(self._hosts)
+
+        try:
+            self.run()
+        except CommandFailure as details:
+            self.log.info("<SERVER> Exception occurred: %s" % details)
+            # Kill the subprocess, anything that might have started
+            self.kill()
+            raise ServerFailed(
+                "Failed to start server in {} mode.".format(
+                    self.runner.job.mode))
+
+        if self.runner.job.yaml_params.is_nvme() or \
+           self.runner.job.yaml_params.is_scm():
+            # Setup the hostlist to pass to dmg command
+            servers_with_ports = [
+                "{}:{}".format(host, self.runner.job.yaml_params.port)
+                for host in self._hosts]
+
+            self.log.info("Formatting hosts: <%s>" % self._hosts)
```

Comment:
(pylint-logging-not-lazy) Specify string format arguments as logging function parameters